x/market/types/deal.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/x/market/util/address"
+	"strings"
+)
+
+func NewDealAddress(dealId uint64) sdk.AccAddress {
+	key := append([]byte("deal"), sdk.Uint64ToBigEndian(dealId)...)
+	return address.Module(ModuleName, key)
+}
+
+func AccDealAddressFromBech32(address string) (sdk.AccAddress, error) {
+	if len(strings.TrimSpace(address)) == 0 {
+		return sdk.AccAddress{}, fmt.Errorf("empty address string is not allowd")
+	}
+
+	bech32PrefixAccAddr := sdk.GetConfig().GetBech32AccountAddrPrefix()
+
+	bz, err := sdk.GetFromBech32(address, bech32PrefixAccAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = VerifyDealAddressFormat(bz)
+	if err != nil {
+		return nil, err
+	}
+
+	return sdk.AccAddress(bz), nil
+}
+
+func VerifyDealAddressFormat(bz []byte) error {
+	verifier := sdk.GetConfig().GetAddressVerifier()
+	if verifier != nil {
+		return nil
```

Comment:
```suggestion
		return verifier(bz)
```
Shouldn't we call the `verifier` function? as https://github.com/cosmos/cosmos-sdk/blob/33dbf6a795e68c744a2c312b76d41614b3563c01/types/address.go#L143

Reply:
I agree with your review. I'll replace ```nil``` with ```verfiier(bz)```.